python/mspasspy/algorithms/resample.py

```diff
@@ -229,6 +229,21 @@ def resample(self, mspass_object):
             )
             mspass_object.set_npts(n_resampled)
             mspass_object.dt = self.dt
+            # Check for "sampling_rate" attribute
+            if mspass_object.is_defined("sampling_rate"):
+                sampling_rate = mspass_object["sampling_rate"]
+                # Check if sampling_rate is consistent with 1/dt
+                if abs(sampling_rate - 1.0 / mspass_object.dt) > 1e-6:
+                    # Record inconsistency in error log (elog)
+                    message = "sampling_rate inconsistent with 1/dt; updating to 1/dt"
+                    mspass_object.elog.log_error("TimeSeries2Trace",
+                                                 message,
+                                                 ErrorSeverity.Invalid)
+                # Update sampling_rate to 1/dt
+                mspass_object["sampling_rate"] = 1.0 / mspass_object.dt
+            else:
```

Comment:
The else branch here is not covered by test. Also, I think we need to add tests for the converter, too. We also need to test for the elog here. I am actually not sure whether elog is correctly retained after the conversion to Trace and back.

Reply:
I fixed it and pushed it. I think elog is correctly retained during my tests.